serenity-screenplay-playwright/src/main/java/net/serenitybdd/screenplay/playwright/abilities/BrowseTheWebWithPlaywright.java

```diff
@@ -232,6 +244,29 @@ public void endPerformance(ActorEndsPerformanceEvent performanceEvent) {
     @Subscribe
     public void testFinishes(TestLifecycleEvents.TestFinished testFinished) {
         if (playwright != null) {
+
+            // Take screenshot for failed test
+            BaseStepListener baseStepListener = StepEventBus.getEventBus().getBaseStepListener();
+            if (baseStepListener.currentTestFailed()) {
+                Page currentPage = getCurrentPage();
+                byte[] screenshot = currentPage.screenshot(new Page.ScreenshotOptions().setFullPage(true));
+
+                try {
+                    Path outputDirectory = baseStepListener.getOutputDirectory().toPath();
+                    Path pageSourceFile = Files.createTempFile(outputDirectory, "pagesource", ".txt");
+                    Path screenshotFile = Files.createTempFile(outputDirectory, "screenshot", ".png");
+                    Files.write(pageSourceFile, currentPage.content().getBytes(StandardCharsets.UTF_8));
+                    Files.write(screenshotFile, screenshot);
+
+                    ScreenshotAndHtmlSource screenshotAndHtmlSource = new ScreenshotAndHtmlSource(screenshotFile.toFile(), pageSourceFile.toFile());
+                    baseStepListener.firstFailingStep().ifPresent(
+                            step -> step.addScreenshot(screenshotAndHtmlSource)
+                    );
+                } catch (IOException e) {
+                    Assertions.fail("Failed to take Playwright screenshot", e);
+                }
+            }
+
             playwright.close();
```

Comment:
This feels like the wrong place for the screenshot logic, as it will not work for steps or Screenplay actions - ideally we need to implement a version of the Photographer that works with Playwright.